fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java

```diff
@@ -198,12 +202,50 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 argTypes[1] = getChild(1).getType();
             }
             fn = getBuiltinFunction(fnName.getFunction(), argTypes,
-                    Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+                Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
             if (fn == null) {
                 LOG.warn("fn {} not exists", this.toSqlImpl());
                 throw new AnalysisException(getFunctionNotFoundError(collectChildReturnTypes()));
             }
             fn.setReturnType(getChild(0).getType());
+        } else if ( fnName.getFunction().equalsIgnoreCase("array_last") ) {
+
+            List<Expr> params = new ArrayList<>();
+            for (int i = 0; i <= childSize - 1; ++i) {
+                params.add(getChild(i));
+            }
+            LambdaFunctionCallExpr filterFunc = new LambdaFunctionCallExpr("array_filter", params);
+            //filterFunc.analyzeImpl(analyzer);
+            IntLiteral indexParam = new IntLiteral(-1, Type.INT);
+            params = Lists.newArrayList(filterFunc, indexParam) ;
+
+            Type[] argTypesForCallExpr = new Type[2];
+            argTypesForCallExpr[0] = getChild(0).getType();
+            argTypesForCallExpr[1] = getChild(1).getType();
+            FunctionCallExpr callExpr = new FunctionCallExpr("element_at", params) ;
+            callExpr.analyzeImpl(analyzer);
+
+            this.setFnName(arrayCloneExpr) ;
+
+            this.children.clear() ;
+            this.children.add(callExpr);
+
+            argTypes = new Type[1] ;
+            argTypes[0] = getChild(0).getType();
+
+            fn = getBuiltinFunction(arrayCloneExpr.getFunction(), argTypes, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+            if (fn == null) {
+                LOG.warn("fn element_at not exists");
+                throw new AnalysisException(getFunctionNotFoundError(collectChildReturnTypes()));
+            }
+            fn.setReturnType(getChild(0).getType());
+        } else if ( fnName.getFunction().equalsIgnoreCase("array_clone_expr") ) {
```

Comment:
I think is better to rewrite array_last(lambda,array)--->element_at(array_filter,-1)